docs/advanced_topics/documents/title_generation_on_upload.md

```diff
@@ -32,82 +30,174 @@ See MDN for more information about [custom JavaScript events](https://developer.
 
 ## Code examples
 
+For each example below, create the specified external JavaScript file in your app’s static directory, such as static/js/, and reference it in the wagtail_hooks.py file.
+
+### Document specific methods:
+
 ### Adding the file extension to the start of the title
 
 ```python
 # wagtail_hooks.py
 from django.utils.safestring import mark_safe
-
 from wagtail import hooks
-
+from django.templatetags.static import static
 
 @hooks.register("insert_global_admin_js")
 def get_global_admin_js():
-    return mark_safe(
-    """
-    <script>
-    window.addEventListener('DOMContentLoaded', function () {
-        document.addEventListener('wagtail:documents-upload', function(event) {
-            var extension = (event.detail.filename.match(/\.([^.]*?)(?=\?|#|$)/) || [''])[1];
-            var newTitle = '(' + extension.toUpperCase() + ') ' + (event.detail.data.title || '');
-            event.detail.data.title = newTitle;
-        });
+    static_url = static('js/title_with_extension.js')
+    return mark_safe(f'<script src="{static_url}"></script>')
```

Comment:
Not familiar, but I'll look into it. Might need help deciding a direction for this.

Reply:
Good call, let's use that `format_html` for other code snippet changes also. 

The mark_safe was used to make it so that we didn't have to escape the JS code. 

https://docs.djangoproject.com/en/5.1/ref/utils/#django.utils.html.format_html

Reply:
Alright, I'll use `format_html` for all the methods.